sdk/containerregistry/azure-containers-containerregistry/src/test/java/com/azure/containers/containerregistry/ContainerRegistryClientAsyncIntegrationTests.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.containers.containerregistry;
+
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.HttpClient;
+import com.azure.core.test.implementation.ImplUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import reactor.test.StepVerifier;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.azure.containers.containerregistry.TestUtils.ALPINE_REPOSITORY_NAME;
+import static com.azure.containers.containerregistry.TestUtils.DISPLAY_NAME_WITH_ARGUMENTS;
+import static com.azure.containers.containerregistry.TestUtils.HELLO_WORLD_REPOSITORY_NAME;
+import static com.azure.containers.containerregistry.TestUtils.PAGESIZE_2;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class ContainerRegistryClientAsyncIntegrationTests extends ContainerRegistryClientsTestBase {
+
+    private ContainerRegistryAsyncClient registryClient;
+    private ContainerRepositoryAsyncClient repositoryClient;
+
+    private ContainerRegistryAsyncClient getContainerRegistryAsyncClient(HttpClient httpClient) {
+        return getContainerRegistryBuilder(httpClient).buildAsyncClient();
+    }
+
+    @BeforeAll
+    static void beforeAll() {
+        StepVerifier.setDefaultTimeout(Duration.ofMinutes(30));
```

Comment:
I think we would be better off using JUnit's [timeout annotation](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Timeout.html). Using `StepVerifier.setDefaultTimeout` affects the global state, and while this timeout is very long when running CI this could lead to unforeseen consequences.

@conniey, @srnagar , I know you use this often, what are your thoughts? Another option would be to add utility methods into azure-core-test which would allow test writers to pass their `StepVerifier` and have it apply a [timeout based verify](https://projectreactor.io/docs/test/release/api/reactor/test/StepVerifier.html#verify-java.time.Duration-). This would have the added benefit that we can set different timeouts between CI and integration testing.

Reply:
When you say global you mean the entire test run? Was not aware of it and actually saw this pattern being used in some tests.